pex/environment.py

```diff
@@ -216,7 +211,20 @@ def _update_candidate_distributions(self, distribution_iter):
           self.add(dist)
 
   def can_add(self, dist):
-    return distribution_compatible(dist, self._supported_tags)
+    filename, ext = os.path.splitext(os.path.basename(dist.location))
+    if ext.lower() != '.whl':
```

Comment:
Could we tighten this check at all, e.g. `ext.lower() != .whl and os.path.isdir(..)`? Right now, something like `foo.egg` would return `True`.

Reply:
In fact, consider raising an exception if the `extension != whl and os.path.isfile()`. Perhaps that's overly defensive, but would be nice to fail eagerly if we make programming errors in the future that try passing a non-wheel name to `can_add`.

Reply:
I'll think on this one. The code was just moved, no logic changes. Although the logic you suggest is correct, it's exactly the sort of thing that led to zipped wheel finders hanging around so long (eliminated in #827). When you read a check like that it makes you think it can actually happen but it can't due to a larger systemic invariant (enforced by collusion between resolver and PEXBuilderat build time).